### PR TITLE
git workflow deploy webhook

### LINKED ISSUE #4: Automated deploy to staging & production
## Acceptance
- [ ] There is a way to deploy to a staging site
- [ ] There is a way to deploy to a production site

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Fixes #4 

## Description

- github workflow webhook

## Deployment

Deploys can be accessed here: https://team-wilder-proto.pages.dev/

We have branch previews so we can call that staging. We should get a subdomain for any branch at

```
https://[branch-name].team-wilder-proto.pages.dev/
```
e.g. https://6-css-reset.team-wilder-proto.pages.dev/

@geeksforsocialchange/developers

### CODE REVIEW

**Comment by ivan-kocienski-gfsc** [comment]
Still figuring this out

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+name: Deploy to Cloudflare
+
+on:
+  - push
```
Comment (katjam) [inline .github/workflows/build.yml:4]: I think on merge to main we want a release and on create Pull Request we only need a preview.

I'm not sure how it works with Cloudflare, but in netlify the PR part of that will happen automatically once the repo is wired up.

**Review by katjam — COMMENTED** [review]
I'd have to check the cloudflare docs to see how to configure, but desired behaviour is PR triggers preview build + merge to main triggers release

**Comment by katjam** [comment]
@ivan-kocienski-gfsc Also don;t forget to update the readme.

**Comment by aaaaargZombies** [comment]
According to cloudflare we should get a build when we commit to `main` so we can call that the production site

https://team-wilder-proto.pages.dev/

we have branch previews so we should get a subdomain for any branch at

```
[branch-name].https://team-wilder-proto.pages.dev/
```
eg https://6-css-reset.team-wilder-proto.pages.dev/

**FYI** you will only get that nice comment bot link to click on your PR if a new build is run when a PR is open. this is a bit annoying and not as good as netlify. we could consider adding a prompt to manually add it to the PR template?

File: .github/workflows/build.yml (hunk 1 above)
Comment (ivan-kocienski-gfsc) [reply .github/workflows/build.yml:4]: i am going to look into making this only deploy on modifications to main. is that enough? as this is a super prototype alpha project do we need to worry about secure production branches?

File: README.md [hunk 2]
```diff
@@ -29,8 +29,8 @@ We aim to cover any exposed API in our modules and things with potential to fail
 
 ## Deployment
 
-- When a pull request is created against `main`, [cCc] builds a preview site
-- When code is merged into `main` it is deployed to [current release]([cCc])
+- When a pull request is created against `main`, cloudflare builds a preview site
+- When code is merged into `main` it is deployed to https://team-wilder-proto.pages.dev/
```
Comment (katjam) [inline README.md:33]: Can you also add build status and current release to the top?

**Review by katjam — CHANGES_REQUESTED** [review]

**Comment by ivan-kocienski-gfsc** [comment]
Is there an example repo I can pull this from or some documentation describing how it's done?

My basic google noodling has found only https://github.com/badges/shields/issues/2714

**Review by katjam — APPROVED** [review]
Ace! :robot:

File: README.md [hunk 3]
```diff
@@ -29,8 +29,8 @@ We aim to cover any exposed API in our modules and things with potential to fail
 
 ## Deployment
 
-- When a pull request is created against `main`, [cCc] builds a preview site
-- When code is merged into `main` it is deployed to [current release]([cCc])
+- When a pull request is created against `main`, cloudflare builds a preview site
```
Comment (aaaaargZombies) [inline README.md:30]: I think this is not quite accurate. it's something like.

```md
Cloudflare creates a build each time an update is pushed to the repository. 

- `main` is our production branch and the build resolves to `https://team-wilder-proto.pages.dev/`
- Any other branch will create a build that will resolve to `https://[branch-name].team-wilder-proto.pages.dev/` we use this for staging.
```

**Review by aaaaargZombies — CHANGES_REQUESTED** [review]
There's one change I'd like to see, I've included some suggested text.

Did Katja suggest documenting the settings somehow, in an image or link to the part of the admin UI in cloudflare? (because we can't make a config file)

I also think it would be good to add a prompt to link to the branch deploy url in a PR description because the bot that cloudflare provides is very flaky. I will leave this up to you as it's a nice to have not a blocker for me.

**Pull request merged**